### PR TITLE
Upgrade frontend

### PR DESCRIPTION
**Pull request by michael-markl**
Upgrades Flutter to v3.7.
Make sure to run `fvm install` after this PR is merged

Some smoke testing on iOS would be great!

### CODE REVIEW

**Review by sarahsporck — COMMENTED** [review]
When running flutter pub get the lock file changes for me. Something related to `mobile_scanner 3.0.0 from git https://github.com/michael-markl/mobile_scanner.git at 4bdbbc (was 3.0.0 from git https://github.com/michael-markl/mobile_scanner.git at 3a4f39)`
Also sth. is not right with the clusters. They seem to overlap: 


Also the category filters don't seem to work:

**Comment by michael-markl** [comment]
> When running flutter pub get the lock file changes for me. Something related to mobile_scanner 3.0.0 from git https://github.com/michael-markl/mobile_scanner.git at 4bdbbc (was 3.0.0 from git https://github.com/michael-markl/mobile_scanner.git at 3a4f39)

It seems you're right! We should adjust the CI to fail in that case...

>  Also sth. is not right with the clusters. They seem to overlap:

The clusters on your screesnhots are the "normal" ones that we have for Nürnberg when you zoom out a lot. Maybe you mixed up project configs somehow in your build? 🤔 

> Also the category filters don't seem to work:

Not sure why this does not work... Again it seems to me like a setup issue 🤔  I will check again once I have time whether I can reproduce this.

File: frontend/lib/themes.dart [hunk 1]
```diff
@@ -11,14 +11,14 @@ ThemeData get lightTheme {
       primary: primaryColor,
       secondary: primaryColor,
       background: Colors.white,
-      surfaceVariant: const Color(0xffededed),
+      surfaceVariant: const Color(0xffefefef),
```
[the remark this answers is not in the document]
Comment (michael-markl) [reply frontend/lib/themes.dart:14]: I tred to improve the contrast a bit (especially for Nürnberg's primary color). It's not a huge change though :D

**Comment by sarahsporck** [comment]
> The clusters on your screesnhots are the "normal" ones that we have for Nürnberg when you zoom out a lot. Maybe you mixed up project configs somehow in your build? 🤔

Hmm, I guess you're right. I also had the feeling sth. was off. I'll take another look at my setup.

**Comment by sarahsporck** [comment]
Okay I got it working. Not sure what didn't work though... I'll update the pods for you.

**Comment by michael-markl** [comment]
Thanks! While you're at it, could you also update the lock file for flutter pub?

**Comment by michael-markl** [comment]
@sarahsporck Did you also try scanning a code on iOS? If yes, I think we can merge this (unless you have other questions/concerns) :D

**Comment by sarahsporck** [comment]
> @sarahsporck Did you also try scanning a code on iOS? If yes, I think we can merge this (unless you have other questions/concerns) :D

Sadly, I don't have an iOS device of my own, so far. And when i tried installing it on the device in the projektraum I had some trouble with provisioning profiles. However, @f1sh1918  can test it when he comes back from vacation on monday :)

**Review by f1sh1918 — APPROVED** [review]
Tested on ios real device. no issues found. nice :)

**Pull request merged**